File: week4/cpp/simplify_add.cpp
#include <bits/stdc++.h> // Include all standard C++ libraries
using namespace std; // Use the standard namespace

void solve(){
	int n; cin>>n;
	string s;	cin>>s;

	string temp = "";
	for(int i=0; i<n; i+=2){

		temp += s[i];
	}
	sort(temp.begin(), temp.end());

	for(int i=0; i<n; i+=2){

		s[i] = temp[i/2];
	}

	cout<<s<<"\n";
}

int main(){
	
	int tstCnt; // Declare an integer variable for the number of test cases

	cin>>tstCnt; // Read the number of test cases from the input

	while(tstCnt--){ // Loop for each test case

		solve(); // Call the solve function

	}

	return 0;
}

File: week4/Py/simplify_add.py
t=int(input())
for _ in range(t):
    n=int(input())
    arr=[x for x in input()]
    nums=[arr[i] for i in range(0,n,2)]
    nums.sort()
    soln=""
    for i in range(0,n-1,2):
        soln+=nums[i//2]
        soln+=arr[i+1]
    soln+=nums[-1] 
    print(soln)
    
